### PR TITLE
Make sure to set the correct cookie attributes when serving from loca…

### PR DESCRIPTION
**Pull request by koenvo**
…lhost

### CODE REVIEW

File: solara/server/flask.py [hunk 1]
```diff
@@ -220,9 +221,20 @@ def read_root(path):
     if not allowed():
         return login()
 
+    samesite = "lax"
```
Comment (koenvo) [inline solara/server/flask.py:224]: I added some duplicate code here. What would be a good place to store this, if you like it to be refactored. Other option might be to test only the `starlette` server and don't touch the `flask` part.

Comment (maartenbreddels) [reply solara/server/flask.py:224]: It’s ok, since we test it, unless you see an elegant way

**Comment by maartenbreddels** [comment]
Very cool, and high quality pr 😍
Do the test really test it since it is the same server and port? If you revert the changes, does the test fail?

**Comment by koenvo** [comment]
> Very cool, and high quality pr 😍 Do the test really test it since it is the same server and port? If you revert the changes, does the test fail?

You're welcome. Happy to contribute!


Good call here. Normally I would write the failing test first 🙃  

Output with the changes reverted:
```bash
FAILED tests/integration/server_test.py::test_run_in_iframe[flask-chromium] - playwright._impl._api_types.TimeoutError: Timeout 18000ms exceeded.
FAILED tests/integration/server_test.py::test_run_in_iframe[starlette-chromium] - playwright._impl._api_types.TimeoutError: Timeout 18000ms exceeded.
```

**Comment by maartenbreddels** [comment]
I did some local testing, and managed to pass the test after disabling your feature. I think it is because it is running on the same server.

Then I made it run on a different port, and it still passes the test (I can see with --headed and PWDEBUG=1 that the page really loads in the iframe). If I test this with `python http.server` like in the docs, it *does* fail.

I think there is still something happening I don't understand.

**Comment by maartenbreddels** [comment]
e.g. try changing localhost to xlocalhost in the check of starlette and flask and the test pass 🤷

**Comment by maartenbreddels** [comment]
Ah, we can use `    page_session.context.clear_cookies()` for this.



> 2\. `http://localhost` are considered cross-origins which is good for the test.

I didn't know! Where did you find this?


Ok, I can confirm that now with clearing the cookies, it fails without your changes.

**Comment by koenvo** [comment]
> Ah, we can use ` page_session.context.clear_cookies()` for this.
Right! I was thinking about restarting the entire webserver but this is what I was actually looking for.

> 
> > 2. `http://localhost` are considered cross-origins which is good for the test.
> 
> I didn't know! Where did you find this?
> 
Test it and use `PWDEBUG` to see if the cookies get accepted. By default the `page_session.url` is `about:blank`

File: tests/integration/server_test.py [hunk 2]
```diff
@@ -124,3 +124,34 @@ def test_from_thread_two_users(browser: playwright.sync_api.Browser, solara_serv
...
         page2.close()
+
+
+@solara.component
+def StaticPage():
+    return w.Label(value="Hello world")
+
+
+def test_run_in_iframe(page_session: playwright.sync_api.Page, solara_server, solara_app, extra_include_path, tmpdir):
+    tmpdir.join("index.html").write(
```
[the remark this answers is not in the document]
Comment (maartenbreddels) [reply tests/integration/server_test.py]: thanks!

**Comment by maartenbreddels** [comment]
Great work, thanks Koen!

**Pull request merged**